jwst/extract_1d/ifu.py

```diff
@@ -495,31 +507,56 @@ def extract_ifu(input_model, source_type, extract_params):
                 log.warning("Background annulus has no area, so background "
                             "subtraction will be turned off. %g" ,k)
                 subtract_background_plane = False
-        del temp
 
         npixels[k] = aperture_area
-        npixels_annulus[k] = 0.0
+        npixels_bkg[k] = 0.0
         if annulus is not None:
-            npixels_annulus[k] = annulus_area
+            npixels_bkg[k] = annulus_area
         # aperture_photometry - using data
 
         phot_table = aperture_photometry(data[k, :, :], aperture,
                                          method=method, subpixels=subpixels)
         temp_flux[k] = float(phot_table['aperture_sum'][0])
+
+        # Point source type of data with defined annulus size
         if subtract_background_plane:
             bkg_table = aperture_photometry(data[k, :, :], annulus,
                                             method=method, subpixels=subpixels)
             background[k] = float(bkg_table['aperture_sum'][0])
             temp_flux[k] = temp_flux[k] - background[k] * normalization
 
+        # Extended source data - background determined from sigma clipping
+        if source_type == 'EXTENDED':
+            bkg_data = data[k, :, :]
+            # pull out the data where weight map => 1. We do not want to use
+            # the edge data that is zero to define the statistics on clipping
+            bkg_stat_data = bkg_data[temp_weightmap == 1]
```

Comment:
I will clean up the comment to make it clear. weight_map > = 1 is the same as temp_weightmap = 1 because of how  temp_weightmap was defined
temp_weightmap = weightmap[k,:,:]
        temp_weightmap[temp_weightmap>1] = 1